### PR TITLE
Cluster_builder implementation

### PR DESCRIPTION
**Pull request by dhavalpatel624624**
Implementation logic for cluster_builder which clusters all the possible models and their results and selects the top ones by running a Kmeans on the clusters. 
Plotting is only included for wss (within groups sum of squares)
Plots for top_solutions_errors_plot, top_solutions_rois_plot, and clusters_confidence_interval_plot are PENDING

Testing Validation:
WSS Plot:
image

Confidence interval df and clusters mean values:
image

cluster_results, cluster data and the top solutions/models from the output:
image

### CODE REVIEW

File: python/src/robyn/modeling/clustering/cluster_builder.py [hunk 1]
```diff
@@ -1,10 +1,38 @@
...
+
+import re
+import numpy as np
 import pandas as pd
-from robyn.modeling.clustering.clustering_config import ClusteringConfig
-from robyn.modeling.entities.clustering_results import ClusteredResult
+from robyn.modeling.clustering.clustering_config import (
+    ClusterBy,
+    ClusteringConfig,
+    DependentVarType,
```
Comment (alxlyj) [inline python/src/robyn/modeling/clustering/cluster_builder.py:14]: shouldn't this be from robyn.data.entities.enums import DependentVarType?

File: python/src/robyn/tutorials/tutorial7_clustering.ipynb
Comment (alxlyj) [inline python/src/robyn/tutorials/tutorial7_clustering.ipynb:1]: could you address this?

A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

**Review by alxlyj — APPROVED** [review]
few nits

File: python/src/robyn/modeling/clustering/cluster_builder.py (hunk 1 above)
Comment (dhavalpatel624624) [reply python/src/robyn/modeling/clustering/cluster_builder.py:14]: good catch, I'll address it. I just took what it auto suggested.

File: python/src/robyn/tutorials/tutorial7_clustering.ipynb
Comment (dhavalpatel624624) [reply python/src/robyn/tutorials/tutorial7_clustering.ipynb:1]: Yep, will address it in the next PR :)

**Pull request merged**